DOC-994 Istio ingress configuration

kaitlynmichael opened the pull request:
DOC-994
Staging: https://docs.redis.com/staging/jira-doc-994/kubernetes/re-databases/ingress_routing_with_istio/

based off: https://redislabs.atlassian.net/wiki/spaces/K8s/pages/3199008780/Istio+as+an+Ingress+controller+-+Installation+configuration+and+accessing+a+Redis+Enterprise+database+from+outside+the+K8S+cluster

laurentdroin commented:
The IP addresses have an invalid format. For example, 12.345.678.910 will strike people as invalid since values would be <= 255 for ipv4. I know that it shouldn't matter since they are not supposed to be used as-is anyway but still, I can guarantee this will annoy many nerds, not just me :-)

laurentdroin commented:
v1alpha3 is old, let's not use it for the istio resources (I know it's all over the official istio examples but still).
Instead, we should go with "networking.istio.io/v1beta1"

laurentdroin commented:
I think that the following:

"The gateway’s metadata name must be similar to the gateway’s spec name (redis-gateway in this example)."

is misplaced. It would make some sense if the Gateway came second in the list of Istio resources but it comes first.

I believe that what we mean is the following:

The 'gateway listed in the "gateways" spec of the VirtualService resource must correspond to the name of the Gateway resource'
And for this to make sense, we need to have introduced both resources (i.e. we can't put this text in between resources).

Or something similar to the above... We just need to explain the link between the 2 resources.

laurentdroin commented:
I don't know if we are making it clear that with the VirtualService example, we are actually creating both a route to contact the API server and a route to contact one of the databases. A reader might wonder where is "db1" coming from so we may want to explain this a bit more.
And if we are just explaining how to access the databases, do we really need to bother with a route for the API?

bascar commented:
> v1alpha3 is old, let's not use it for the istio resources (I know it's all over the official istio examples but still). Instead, we should go with "networking.istio.io/v1beta1"

Have we tested against v1beta1?

laurentdroin commented:
> > v1alpha3 is old, let's not use it for the istio resources (I know it's all over the official istio examples but still). Instead, we should go with "networking.istio.io/v1beta1"
> 
> Have we tested against v1beta1?

Not that it's official, but I have.

laurentdroin commented:
@bascar Looks like there is no difference between v1alpha3 and v1beta1 - just better to go with the latest version.

See: https://discuss.istio.io/t/is-api-version-networking-istio-io-v1alpha3-considered-to-be-deprecated-on-1-5-0/7278

kaitlynmichael commented:
New edits from feedback: 

- new example IPs
- replaced v1alpha3 with v1beta1
- moved gateway metadata explanation after virtual service yaml
- added note after vs example about creating route for API and db

rrelledge commented on content/kubernetes/re-databases/ingress_routing_with_istio.md:
````diff
@@ -0,0 +1,155 @@
+    istio-ingressgateway   LoadBalancer   10.34.67.89   10.145.78.91   15021:12345/TCP,80:67891/TCP,443:23456/TCP,31400:78901/TCP,15443:10112/TCP   3h8m
+    ```
+
+1. Create a DNS entry that resolves your chosen database hostname (or a wildcard `*` followed by your domain) to the Istio `EXTERNAL-IP`. This hostname is what you will use to access your database from outside the cluster.
````
```suggestion
1. Create a DNS entry that resolves your chosen database hostname (or a wildcard `*` followed by your domain) to the Istio `EXTERNAL-IP`. Use this hostname to access your database from outside the cluster.
```

rrelledge commented on content/kubernetes/re-databases/ingress_routing_with_istio.md:
```diff
@@ -0,0 +1,155 @@
+
+1. Create a DNS entry that resolves your chosen database hostname (or a wildcard `*` followed by your domain) to the Istio `EXTERNAL-IP`. This hostname is what you will use to access your database from outside the cluster.
+
+    In this example, any hostname ending in `.istio.k8s.my.redisdemo.com` will resolve to the Istio LoadBalancer's external IP of `12.345.678.910`. Substitute your own values accordingly.
```
```suggestion
    In this example, any hostname that ends with `.istio.k8s.my.redisdemo.com` will resolve to the Istio LoadBalancer's external IP of `12.345.678.910`. Substitute your own values accordingly.
```

rrelledge commented on content/kubernetes/re-databases/ingress_routing_with_istio.md:
```diff
@@ -0,0 +1,155 @@
+
+### `VirtualService` custom resource
+
+1. On different namespace than `istio-system`, create the `VirtualService` custom resource file (`redis-vs.yaml`).
```
```suggestion
1. On a different namespace than `istio-system`, create the `VirtualService` custom resource file (`redis-vs.yaml`).
```

rrelledge commented on content/kubernetes/re-databases/ingress_routing_with_istio.md:
```diff
@@ -0,0 +1,155 @@
+
+    This creates both a route to contact the API server and a route to contact one of the databases (`db1` in this example).
+
+    - Replace `.istio.k8s.my.redisdemo.com` with the domain matching your DNS record.
```
```suggestion
    - Replace `.istio.k8s.my.redisdemo.com` with the domain that matches your DNS record.
```

rrelledge commented on content/kubernetes/re-databases/ingress_routing_with_istio.md:
````diff
@@ -0,0 +1,155 @@
+          mode: PASSTHROUGH
+    ```
+
+    - Replace `.istio.k8s.my.redisdemo.com` with the domain matching your DNS record.
````
```suggestion
    - Replace `.istio.k8s.my.redisdemo.com` with the domain that matches your DNS record.
```

rrelledge commented on content/kubernetes/re-databases/ingress_routing_with_istio.md:
```diff
@@ -0,0 +1,155 @@
+
+## Install and configure Istio for Redis Enterprise
+
+1. [Download](https://istio.io/latest/docs/setup/getting-started/) and [install](https://istio.io/latest/docs/setup/getting-started/) Istio (see instructions from Istio's [Getting Started](https://istio.io/latest/docs/setup/getting-started/) guide).
```
The same "Getting Started" link appears three times here. Maybe limit the link to a single occurrence or provide more specific links for download and install?

rrelledge commented on content/kubernetes/re-databases/ingress_routing_with_istio.md:
```diff
@@ -0,0 +1,155 @@
+
+To [test your external access]({{<relref "/kubernetes/re-databases/set-up-ingress-controller.md">}}) to the database, you need a client that supports [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) and [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication).
+
+See [Test your access with Openssl]({{<relref "/kubernetes/re-databases/set-up-ingress-controller.md#test-your-access-with-openssl">}}) or [Test your access with Python]({{<relref "http://localhost:1313/kubernetes/re-databases/set-up-ingress-controller.md#test-your-access-with-python">}}) for more info.
```
```suggestion
See [Test your access with Openssl]({{<relref "/kubernetes/re-databases/set-up-ingress-controller#test-your-access-with-openssl">}}) or [Test your access with Python]({{<relref "/kubernetes/re-databases/set-up-ingress-controller#test-your-access-with-python">}}) for more info.
```

rrelledge reviewed (approved):
Looks good to me. I just left a few suggestions.
